component/core/src/main/java/io/meeds/social/space/service/SpaceServiceImpl.java

```diff
@@ -1405,4 +1408,86 @@ private String buildPrettyName(String... names) {
             .orElseThrow();
   }
 
+  private void checkSubspaceCreationAllowed(Space space, long parentSpaceId) throws SpaceException {
+    if (parentSpaceId <= 0) {
+      return;
+    }
+    Space parentSpace = getSpaceById(parentSpaceId);
+    SpaceTemplate parentTemplate = spaceTemplateService.getSpaceTemplate(parentSpace.getTemplateId());
+    if (parentTemplate == null) {
+      throw new SpaceException(Code.UNKNOWN_SPACE_TEMPLATE,
+                               String.format("Unknown parent space template for space %s", parentSpace.getDisplayName()));
+    }
+
+    Integer subspacesMaxLimit = parentTemplate.getSubspacesMaxLimit();
+    List<String> allowedSubspaceTemplates = parentTemplate.getAllowedSubspaceTemplates();
+
+    SpaceFilter spaceFilter = new SpaceFilter();
+    spaceFilter.setParentSpaceId(parentSpaceId);
+    ListAccess<Space> existingSubspacesAccess = getAllSpacesByFilter(spaceFilter);
+    Space[] existingSubspaces;
+    try {
+      existingSubspaces = existingSubspacesAccess.load(0, -1);
```

Comment:
Use `getSize` rather thn loading all subspaces

```suggestion
      subspacesCount = existingSubspacesAccess.getSize();
```